agroal-test/src/test/java/io/agroal/test/narayana/CommitMarkableResourceTest.java

```diff
@@ -0,0 +1,246 @@
+import jakarta.transaction.HeuristicMixedException;
+import jakarta.transaction.HeuristicRollbackException;
+import jakarta.transaction.NotSupportedException;
+import jakarta.transaction.RollbackException;
+import jakarta.transaction.SystemException;
+import jakarta.transaction.TransactionManager;
+import jakarta.transaction.TransactionSynchronizationRegistry;
+import org.jboss.tm.ConnectableResource;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.naming.spi.InitialContextFactory;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Hashtable;
+import java.util.logging.Logger;
+
+import static io.agroal.test.AgroalTestGroup.FUNCTIONAL;
+import static io.agroal.test.AgroalTestGroup.TRANSACTION;
+import static java.text.MessageFormat.format;
+import static java.util.List.of;
+import static java.util.logging.Logger.getLogger;
+import static javax.naming.Context.INITIAL_CONTEXT_FACTORY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+@Tag( FUNCTIONAL )
+@Tag( TRANSACTION )
+public class CommitMarkableResourceTest {
+
+    private static final String JNDI_NAME = CommitMarkableResourceTest.class.getName();
+    private static final Logger logger = getLogger( CommitMarkableResourceTest.class.getName() );
+
+    @BeforeAll
+    @SuppressWarnings( "AccessOfSystemProperties" )
+    static void setup() throws NamingException, ClassNotFoundException {
+        System.setProperty( INITIAL_CONTEXT_FACTORY, SimpleMapContextFactory.class.getName() );
+    }
+
+    @AfterAll
+    static void teardown() throws NamingException {
+        System.clearProperty( INITIAL_CONTEXT_FACTORY );
+    }
+
+    // --- //
+
+    @ParameterizedTest
+    @DisplayName( "Test Last Commit Resource Optimization (LRCO)" )
+    @ValueSource( booleans = {false, true} )
+    void lastResourceCommitOptimizationTest(boolean performImmediateCleanupOfBranches) throws SQLException {
+        jtaPropertyManager.getJTAEnvironmentBean().setLastResourceOptimisationInterface( ConnectableResource.class );
+        jtaPropertyManager.getJTAEnvironmentBean().setPerformImmediateCleanupOfCommitMarkableResourceBranches( performImmediateCleanupOfBranches );
+        jtaPropertyManager.getJTAEnvironmentBean().setCommitMarkableResourceJNDINames( of( JNDI_NAME ) );
+
+        if ( performImmediateCleanupOfBranches ) {
+            logger.info( "Performing immediate cleanup of branches is not supported" );
```

Comment:
@jesperpedersen please comment in [AG-214](https://issues.redhat.com/projects/AG/issues/AG-214) if you agree.